### PR TITLE
Fix dynamic component to support embroider-optimised

### PR DESCRIPTION
**Pull request by mansona**
This PR was a long time coming and has been the result of many experiments and I'm so happy that it's finally ready to be merged 🎉

### CODE REVIEW

**Comment by mansona** [comment]
I meant to add this as a draft but I can't see the "make draft" button anywhere 🤔  🤷

**Comment by mansona** [comment]
eh... I did it 😳 I really did it 🎉 it finally works 🎉 I've never been so happy to see a full green CI 😍

File: package.json [hunk 1]
```diff
@@ -30,18 +30,19 @@
...
+    "@embroider/macros": "^1.6.0",
+    "@embroider/util": "^1.0.0",
     "@glimmer/component": "^1.0.4",
     "broccoli-funnel": "^3.0.3",
     "broccoli-merge-trees": "^4.2.0",
     "broccoli-static-site-json": "^4.4.1",
-    "ember-auto-import": "^1.11.3 || ^2.0.0",
-    "ember-cli-babel": "^7.26.6",
-    "ember-cli-htmlbars": "^5.7.1",
+    "ember-auto-import": "github:mansona/ember-auto-import#non-workspace",
```
Comment (elwayman02) [inline package.json:39]: I assume landing this on a published version will be considered a blocker for rolling out this PR?

**Review by elwayman02 — COMMENTED** [review]
Fixes look great! Nice job. Looking forward to getting this out once all the upstream forks are merged and published to support this.

File: package.json (hunk 1 above)
Comment (mansona) [reply package.json:39]: Yes absolutely! When talking to Ed about https://github.com/ef4/ember-auto-import/pull/530 one of the things I was to check was to see if it "solves my problem" 

I'll also probably end up splitting this PR into a few smaller more focused ones 👍

**Pull request merged**